client-haskell/src/Icepeak/Client.hs

```diff
@@ -47,12 +49,15 @@ import qualified Network.HTTP.Types.Status as HTTP
 import qualified Network.HTTP.Types.Header as Header
 import qualified Network.HTTP.Types.URI as URI
 
--- | How to connect to Icepeak?
-data Client = Client
-  { clientHost  :: ByteString
-  , clientPort  :: Word16
-  , clientToken :: Maybe ByteString
-  }
+-- | Details paired with HTTP manager for making requests.
+data Client = Client HTTP.Manager Details
+
+-- | Client connection details: how to connect to Icepeak?
+data Details = Details
```

Comment:
Nit: `Details` is a very generic name, something like `ConnectionSettings` would be more descriptive.

Reply:
Renamed to Config.